src/ring_theory/localization.lean

```diff
@@ -954,6 +954,18 @@ def integral_domain_localization {M : submonoid A} (hM : M ≤ non_zero_divisors
   integral_domain (localization M) :=
 (localization.of M).integral_domain_of_le_non_zero_divisors hM
 
+/--
+The localization of an integral domain at the complement of a prime ideal is an integral domain.
+-/
+instance integral_domain_of_local_at_prime {P : ideal A} (hp_prime : P.is_prime) :
```

Comment:
I'm not an expert on typeclasses, but I think the rule of thumb is that all parameters of an instance should be either `{implicit}` or `[typeclasses]`. And `ideal.is_prime` is a typeclass, so it should be:
```suggestion
instance integral_domain_of_local_at_prime {P : ideal A} [P.is_prime] :
```

Reply:
I'm not sure either, but I think that it might be cumbersome to use instances if you want to consider multiple localizations of the same ring (although I'm not sure if there are enough interesting uses to warrant this).

Reply:
Aha, it looks like the `is_prime` instance is already found by unification with `localization.at_prime`. So it should indeed work fine like you had originally.